teps/0019-other-arch-support.md

```diff
@@ -0,0 +1,424 @@
+    env:
+      - name: KUBECONFIG
+        value: $(workspaces.kubeconfig-dir.path)/kubeconfig
+```
+
+#### Example 2: Run e2e tests
+
+Task is started from amd64 Tekton cluster and runs the e2e tests on non-amd64
+cluster via `-kubeconfig` specification. kubeconfig file to access non-amd64
+cluster is stored at kubeconfig-dir workspace.
+`TEST_RUNTIME_ARCH` is equal to target native architecture(`arm64`, `s390x`,
+`ppc64le`, etc)
+```
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: e2e-tests
+spec:
+  inputs:
+    params:
+    - name: package
+      description: package (and its children) under test
+    - name: tests-path
+      description: path to the tests within "tests" git resource
+      default: ./test
+    - name: timeout
+      description: timeout for the go test runner
+      default: 20m
+    - name: container-registry
+      description: container registry used to push images during tests (can be local or external)
+    - name: tests-arch
+      description: target hardware architecture for tests
+    resources:
+    - name: plumbing-library
+      type: git
+    - name: tests
+      type: git
+      targetPath: src/$(inputs.params.package)
+  workspaces:
+    - name: kubeconfig-dir
+  steps:
+    - name: run-e2e-tests
+      image: gcr.io/tekton-releases/dogfooding/test-runner:latest
+      workingdir: /workspace/src/$(inputs.params.package)
+      env:
+      - name: REPO_ROOT_DIR
+        value: $(inputs.resources.tests.path)
+      - name: GOPATH
+        value: /workspace
+      - name: KO_DOCKER_REPO
+        value: $(inputs.params.container-registry)
+      command:
+      - /bin/bash
+      args:
+      - -ce
+      - |
+        # Setup
+        source $(inputs.resources.plumbing-library.path)/scripts/library.sh
+        e2e_failed=0
+        header "Running Go e2e tests"
+        if [[ -f $(workspaces.kubeconfig-dir.path)/kubeconfig ]]; then
+          export KUBECONFIG=$(workspaces.kubeconfig-dir.path)/kubeconfig
+        fi
+        export TEST_RUNTIME_ARCH=$(inputs.params.tests-arch)
+        report_go_test -v -count=1 -tags=e2e -timeout=$(inputs.params.timeout) $(inputs.params.tests-path) -kubeconfig $(workspaces.kubeconfig-dir.path)/kubeconfig
+```
+
+### Risks and Mitigations
+
+1. Approach to support several architectures may require multi-arch specific
+knowledge, but usually it doesn't mean that for any new architecture developer
+should have knowledge about this architecture. In general basic
+knowledge about existance of non-amd64 architectures and
+multi-arch images is enough.
+Most of arch-specific challenges are connected to hardware and k8s setup, which
+is out of scope for this TEP.
+
+2. Usually developers test the new code locally only for amd64, so failing tests
```

Comment:
This is a risk that doesn't describe a mitigation. Who's responsible for debugging and fixing test failures that only affect non-amd64 builds? Is that the regular buildcop? Some person/people with arch-specific knowledge?

Reply:
That depends on the approach community will choose to do the builds (based on my opinion):
1) `ko resolve --platform=all` means support for all possible platforms. Here I believe special buildcop(s) with knowledge about non-amd64 architectures and probably access to all possible hardware.
or
2) Only specific platforms are supported with`ko resolve --platform=linux/s390x`,`ko resolve --platform=linux/arm64`, etc. In this case interested party should provide hardware for tests and specialists who will be responsible to fix failures specific to concrete platform.

Reply:
taking into account this [comment](https://github.com/tektoncd/community/pull/211#discussion_r497399021) I'd like to reformulate my suggestion. 
Interested parties should provide hardware for tests and specialists who will be responsible to fix failures for `supported` architectures. 
Other architectures which `might be available` are not supported or tested. Fixes for them can be provided following standard procedure by any interested person.